lighthouse-core/gather/driver.js

```diff
@@ -1145,6 +1145,7 @@ class Driver {
     } else if (waitForLoad) {
       const passConfig = /** @type {Partial<LH.Config.Pass>} */ (passContext.passConfig || {});
       let {pauseAfterLoadMs, networkQuietThresholdMs, cpuQuietThresholdMs} = passConfig;
+      let pierceIframes = passContext.settings && passContext.settings.pierceIframes;
```

Comment:
it might make more sense to me if we could just set the value of `waitForFCP` based on `pierceIframes` before this function, i.e. on line 68

```
waitForFCP: passContext.passConfig.recordTrace && !passContext.settings.pierceIframes,